core/primitives/src/block.rs

```diff
@@ -745,75 +738,99 @@ impl Block {
     }
 }
 
-pub enum ChunksCollection<'a> {
-    V1(Vec<ShardChunkHeader>),
-    V2(&'a [ShardChunkHeader]),
-}
-
-pub struct VersionedChunksIter<'a> {
-    chunks: &'a [ShardChunkHeader],
-    curr_index: usize,
-    len: usize,
+#[derive(Clone)]
+pub enum MaybeNew<'a, T> {
+    New(&'a T),
+    Old(&'a T),
 }
 
-impl<'a> VersionedChunksIter<'a> {
-    fn new(chunks: &'a [ShardChunkHeader]) -> Self {
-        Self { chunks, curr_index: 0, len: chunks.len() }
+fn annotate_chunk(
+    chunk: &ShardChunkHeader,
+    block_height: BlockHeight,
+) -> MaybeNew<ShardChunkHeader> {
+    if chunk.is_new_chunk(block_height) {
+        MaybeNew::New(chunk)
+    } else {
+        MaybeNew::Old(chunk)
     }
 }
 
-impl<'a> Iterator for VersionedChunksIter<'a> {
-    type Item = &'a ShardChunkHeader;
-
-    fn next(&mut self) -> Option<Self::Item> {
-        if self.curr_index < self.len {
-            let item = &self.chunks[self.curr_index];
-            self.curr_index += 1;
-            Some(item)
-        } else {
-            None
-        }
-    }
+pub enum ChunksCollection<'a> {
+    V1(Vec<ShardChunkHeader>),
+    V2(&'a [ShardChunkHeader]),
```

Comment:
Can we also consolidate V1 and V2 and only have a single version (since this is not a protocol schema struct, we can update the structure together with the code using it)?